oak_remote_attestation_verification/src/verifier.rs

```diff
@@ -70,25 +79,161 @@ impl AttestationVerifier for InsecureAttestationVerifier {
+            });
+        // TODO(#3641): Set other verifiers, e.g., reference value verifiers.
+        let verifiers = Vec::new();
+        Self {
+            verifiers,
+            binary_verifiers,
+            default_binary_verifier,
+        }
+    }
+}
+
+impl AttestationVerifier for AttestationLayerVerifier {
+    fn verify(
+        &self,
+        evidence: &AttestationEvidence,
+        endorsement: &AttestationEndorsement,
+        reference_value: &ReferenceValue,
+    ) -> anyhow::Result<()> {
+        for verifier in &self.verifiers {
+            verifier.verify(evidence, endorsement, reference_value)?;
+        }
+        // TODO(#3641): Update accordingly once #4074 is completed.
+        for verifiers in self.binary_verifiers.values() {
+            for verifier in verifiers {
+                verifier.verify(evidence, endorsement, reference_value)?;
+            }
+        }
+        //  TODO(#3641): Run the following for binaries not specified in `binary_verifiers`.
+        if let Some(verifiers) = &self.default_binary_verifier {
+            for verifier in verifiers {
+                verifier.verify(evidence, endorsement, reference_value)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Creates a list of AttestationVerifiers based on the given verification options.
+fn create_transparency_verifier(
+    opts: &TransparencyVerificationOptions,
+) -> Vec<Box<dyn AttestationVerifier>> {
+    let mut verifiers: Vec<Box<dyn AttestationVerifier>> = Vec::new();
+    verifiers.push(Box::new(EndorsementStatementVerifier {}));
+    if let Some(VerificationData(data)) = &opts.rekor_entry_verification {
+        verifiers.push(Box::new(RekorLogEntryVerifier {
+            base64_pem_encoded_rekor_public_key: data.base64_pem_encoded_rekor_public_key.clone(),
+            base64_pem_encoded_endorser_public_key: data
+                .base64_pem_encoded_endorser_public_key
+                .clone(),
+        }))
+    }
+    verifiers
+}
+
+/// AttestationVerifier for verifying the content of an endorsement statement and checking that it
+/// has the same binary digest as the measurement.
+pub struct EndorsementStatementVerifier {
+    // TODO(#3641): Additional info may be required here to correctly extract the binary digest from
+    // the given evidence.
+}
+
+impl AttestationVerifier for EndorsementStatementVerifier {
+    fn verify(
+        &self,
+        evidence: &AttestationEvidence,
+        endorsement: &AttestationEndorsement,
+        _reference_value: &ReferenceValue,
+    ) -> anyhow::Result<()> {
+        if let Some(binary_attestation) = &endorsement.binary_attestation {
+            verify_endorsement_statement(
+                &binary_attestation.endorsement_statement,
+                &get_binary_digest(evidence),
+            )?;
+        }
+        Ok(())
+    }
+}
+
+fn get_binary_digest(_evidence: &AttestationEvidence) -> Vec<u8> {
```

Comment:
I don't think this signature works. It does not specify which binary's digest must be retrieved from the attestation evidence.